### PR TITLE
Changes in schema model for new translation layer

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
This PR includes changes required in schema model for the translation layer:

* Update .eslintrc with `no-unused-vars` rule
* Update customResolverAnnotation to generate resolveTree
* Add some utility functions to some adapters needed in translation layer

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3973.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 39 | 43 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 74 | 92 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 48 | 56 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3541 | 3572 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3066 | 3281 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 42 | 51 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 76 | 78 | 2174 |
|  aggregations.DeeplyNestedCount | 13070333 | 13070331 | 4633 | 4710 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 172 | 184 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 63 | 92 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 199 | 212 | 3003 |
|  connections.Connection | 14082 | 14082 | 68 | 79 | 2174 |
|  connections.NestedConnection | 45472 | 45472 | 117 | 120 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 67 | 77 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 100 | 110 | 2174 |
|  create.SimpleMutation | 7 | 7 | 56 | 101 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 32 | 35 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 336 | 340 | 1040 |
|  delete.NestedDeleteInUpdate | 19542 | 19542 | 184 | 205 | 1179 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 41 | 68 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 109 | 106 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 45 | 56 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 68 | 67 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 23 | 35 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 55 | 51 | 2174 |
|  query.QueryWhere | 9712 | 9705 | 41 | 43 | 2166 |
|  query.SimpleQueryWithNestedWhere | 9890 | 9883 | 60 | 83 | 2166 |
|  query.Nested | 10096041 | 10096041 | 10271 | 10315 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 9763 | 9511 | 2004000 |
|  query.OrFilterOnRelationships | 42400 | 41338 | 224 | 199 | 1632 |
|  query.OrFilterOnRelationshipsAndNested | 36128 | 34582 | 236 | 313 | 1632 |
|  query.QueryWithNestedIn | 13627 | 13481 | 61 | 92 | 1291 |
|  query.NestedConnectionWhere | 9834 | 9834 | 57 | 81 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9851 | 9992 | 85 | 104 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 18996 | 19206 | 137 | 165 | 1552 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 166 | 230 | 1134 |
|  query.Fulltext | 80 | 80 | 35 | 39 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 47 | 60 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 73 | 84 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14464 | 14526 | 101 | 115 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 59 | 95 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 95 | 109 | 4516 |
|  sorting.SortWithTopLevelCypher | 3396 | 3396 | 73 | 87 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 71 | 94 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 69 | 121 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 358 | 357 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 288 | 319 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 99 | 141 | 2002 |

</details>
        
Old Schema Generation: 40.250s
Schema Generation: 40.380s
Old Subgraph Schema Generation: 49.922s
Subgraph Schema Generation: 47.817s

File: packages/graphql/src/schema-model/entity/model-adapters/ConcreteEntityAdapter.ts [hunk 1]
```diff
@@ -159,6 +159,24 @@ export class ConcreteEntityAdapter {
...
 
+    public getRelayId(): AttributeAdapter | undefined {
+        // TODO: make this O(1) or memoize
+        for (const attr of this.attributes.values()) {
+            if (attr.annotations.relayId) {
+                return attr;
+            }
+        }
+        return undefined;
+    }
```
Comment (MacondoExpress) [inline packages/graphql/src/schema-model/entity/model-adapters/ConcreteEntityAdapter.ts:170]: You have access to `globalIdField`, which was introduced by the schema generator changes
```suggestion
```

File: packages/graphql/src/schema-model/entity/model-adapters/InterfaceEntityAdapter.ts [hunk 2]
```diff
@@ -42,6 +42,15 @@ export class InterfaceEntityAdapter {
         this.initConcreteEntities(entity.concreteEntities);
     }
 
+    public findAttribute(name: string): AttributeAdapter | undefined {
+        return this.attributes.get(name);
+    }
+
+    // RelayId not supported on interfaces
+    public getRelayId(): AttributeAdapter | undefined {
```
Comment (MacondoExpress) [inline packages/graphql/src/schema-model/entity/model-adapters/InterfaceEntityAdapter.ts:50]: Let's keep it consistent with the suggestion in the ConcreteEntityAdapter

**Review by MacondoExpress — APPROVED** [review]

**Pull request merged**